src/stream/gst/pipeline_builder.rs

```diff
@@ -88,7 +88,24 @@ impl Pipeline {
                 }
             },
             VideoSourceType::Local(local_device) => {
-                format!("v4l2src device={}", &local_device.device_path)
+                match &local_device.typ {
+                    crate::video::video_source_local::VideoSourceLocalType::Usb(_) => {
+                        format!("v4l2src device={}", &local_device.device_path)
+                    }
+                    // For raspberry pi cameras (in Legacy Mode), rpicamsrc
+                    // is used to be used insted v4l2src as a workaround to
+                    // prevent Kernel Oops caused by setting a unsupported but
+                    // wrongly stated as supported by v4l2 Frame Size and/or
+                    // FPS.
+                    crate::video::video_source_local::VideoSourceLocalType::LegacyRpiCam(_) => {
+                        format!("rpicamsrc camera-number=0")
+                    }
+                    typ => {
+                        return Err(SimpleError::new(format!(
+                            "Unsuported VideoSourceLocal: {typ:#?}."
+                        )))
+                    }
```

Comment:
note that this changes the current logic, but all current supported cameras still work (USB cameras)

Reply:
Can you explain a bit more how this changes the logic ? Can you also put that in your commit body ?

Reply:
Sure. Originally, any `VideoSourceLocalType` variant - including `Usb` and `Unknown` - would result in a pipeline using `v4l2src`. Now, only the `Usb` variant will, meaning that `Unknown` is not supported. The idea here is to prevent scenarios like when using v4l2src with the Raspberry Pi camera in Legacy Camera mode, which easily causes Kernel Oops/Panic.

note that I intentionally left the tests untouched so we are obligated to have this discussion - the tests are creating a `crate::video::video_source_local::VideoSourceLocalType::Unknown("Potato")` camera, which will not be supported if we choose this change in behavior.

Reply:
@patrickelectric  ping